classification_analysis.ipynb

```diff
@@ -240,29 +323,57 @@
     "                                                                                  tr['start_ts'],\n",
     "                                                                                  tr['end_ts'])\n",
     "                        \n",
-    "                        if section_gt_leg[\"type\"] == \"WAITING\": \n",
+    "                        if section_gt_leg[\"type\"] == \"WAITING\":\n",
     "                            continue\n",
+    "                        \n",
     "                        gts = {'start_ts': section['start_ts'], \n",
     "                               'end_ts': section['end_ts'], \n",
-    "                               'mode': section_gt_leg['mode']}\n",
+    "                               'mode': section_gt_leg['mode']\n",
+    "                               }\n",
+    "\n",
     "                        tr_gts.append(gts)\n",
+    "\n",
+    "                    gt_location_data = get_gt_location_data(FILE_MAPPING[pv], tr['trip_id_base'])\n",
+    "                    run_id = r['trip_run']\n",
+    "                    print(run_id, tr['trip_id'], tr['trip_id'])\n",
+    "                    gt_location_data = gt_location_data.query(\"source == @os\")\n",
+    "                    gt_location_data = gt_location_data.loc[gt_location_data.run == str(run_id), :].reset_index(drop=True, inplace=False)\n",
+    "                    gt_dfs.append(gt_location_data)\n",
+    "\n",
+    "                print(5*'-')\n",
+    "\n",
     "                # now, we build a timeline for each trip\n",
     "                trip = tr.copy()\n",
     "                trip['ss_timeline']  = tr_ss\n",
     "                trip['gts_timeline'] = tr_gts\n",
+    "                trip['gt_location_df'] = gt_dfs\n",
+    "                \n",
     "                trips.append(trip)\n",
+    "    \n",
     "    return trips"
    ]
   },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b0ec3c7e",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "trips = get_trip_ss_and_gts_timeline(pv_la, 'android', 'HAHFDC')"
```

Comment:
> Not all of thee 6 runs (for the case of the LA spec) are read. Instead, only run 0, 1, and 2 are read. Is this expected?

Is this your question about 3 runs? From https://github.com/MobilityNet/mobilitynet.github.io#data-characteristics  (or the paper) it looks like we do indeed have 6 HAHFDC runs for that timeline (so presumably 3 android and 3 ios.

Reply:
Actually, I take that back; there MAHFDC and HAMFDC entries are 3 and 3, respectively, so I don't think we are doubling up on android/ios. So there must be another reason why you only see 3 runs read. Is there another error? Are we skipping for some other reason?

Reply:
were you able to get 6 runs at the end?

Reply:
Yes, we end up with 6 runs.